Mathlib/Util/CompileInductive.lean

```diff
@@ -0,0 +1,179 @@
+/-
+Copyright (c) 2023 Parth Shastri. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Parth Shastri
+-/
+import Lean.Compiler.CSimpAttr
+import Lean.Elab.Predefinition
+
+/-!
+# Define the `#compile inductive` command.
+-/
+
+namespace Mathlib.Util
+
+open Lean
+
+private def replaceConst (old new : Name) (e : Expr) : Expr :=
+  e.replace λ | .const n us => if n == old then some (.const new us) else none | _ => none
+
+open Meta
+
+private def mkFunExts' (xs : Array Expr) (e : Expr) (βfg : Expr × Expr × Expr) : MetaM Expr :=
+  Prod.fst <$> xs.foldrM (init := (e, βfg)) λ x (e, β, f, g) => do
+    let α ← inferType x
+    let f ← mkLambdaFVars #[x] f
+    let g ← mkLambdaFVars #[x] g
+    return (
```

Comment:
I think these parentheses are unnecessary (`return` is a keyword). Or you could use `$`.